### PR TITLE
feat(application): Document how to declare advanced applications.

### PR DESCRIPTION
**Pull request by vhemery**
Start skeleton for Bonita UI Designer Pro.
Detail the declaration in app descriptor (temporaryly in sources).

Relates to [BPM-91](https://bonitasoft.atlassian.net/browse/BPM-91)

[BPM-91]: https://bonitasoft.atlassian.net/browse/BPM-91?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: modules/pages-and-forms/pages/pro-declare-interface-in-bonita.adoc [hunk 1]
```diff
@@ -0,0 +1,27 @@
+= Declare the interface in Bonita
+:page-aliases: ROOT:pro-declare-interface-in-bonita.adoc
+:description: Steps to declare your interface and display it in Bonita.
```
Comment (jeromecambon) [inline modules/pages-and-forms/pages/pro-declare-interface-in-bonita.adoc]:
```suggestion
:description: This page defines the steps to declare your interface and display it in Bonita.
```

File: modules/pages-and-forms/pages/pro-declare-interface-in-bonita.adoc [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+{description}
+
+== Declare the interface in Bonita application descriptor
+
+The xref:applications:application-creation.adoc[Bonita application descriptor] is a file that let you describe all your applications.
+
+To declare a Bonita UI Designer Pro interface in it, you need to edit it in Studio using the "source" tab.
```
Comment (jeromecambon) [inline modules/pages-and-forms/pages/pro-declare-interface-in-bonita.adoc]:
```suggestion
To declare a Bonita UI Designer Pro interface in it, edit it in Studio using the "source" tab.
```

File: modules/pages-and-forms/pages/pro-declare-interface-in-bonita.adoc [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+In this tab, you will be able to declare your application built and packaged with Bonita UI Designer Pro, using auto-completion with a "advancedApplication" XML node.
```
Comment (jeromecambon) [inline modules/pages-and-forms/pages/pro-declare-interface-in-bonita.adoc]:
```suggestion
In this tab, you will be able to declare your application built and packaged with Bonita UI Designer Pro, using auto-completion with a "advancedApplication" XML tag.
```

File: modules/pages-and-forms/pages/pro-declare-interface-in-bonita.adoc [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+image::images/advanced-application-descriptor.png[Application Descriptor Editor]
+
+In this example, the "Advanced Overview" application is declared with node:
```
Comment (jeromecambon) [inline modules/pages-and-forms/pages/pro-declare-interface-in-bonita.adoc]:
```suggestion
In this example, the "Advanced Overview" application is declared with the tag:
```

File: modules/pages-and-forms/pages/bonita-ui-pro.adoc [hunk 5]
```diff
@@ -0,0 +1,18 @@
+
+= Bonita UI Desiger Pro
+:page-aliases: ROOT:bonita-ui-pro.adoc
+:description: Bonita UI Desiger Pro allows to implement "forms" and "pages" in an app, not as standalone elements in a process (paradigm change)
```
Comment (jeromecambon) [inline modules/pages-and-forms/pages/bonita-ui-pro.adoc]:
```suggestion
:description: Bonita UI Designer Pro allows to implement "forms" and "pages" in an application.
They are not anymore standalone elements in a process (paradigm change with legacy Bonita UI Designer).
```

**Review by jeromecambon — COMMENTED** [review]
LGTM, few comments (on the form)

**Review by jeromecambon — APPROVED** [review]
Seems good!

File: modules/pages-and-forms/pages/pro-declare-interface-in-bonita.adoc [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+:description: This page defines the steps to declare your interface and display it in Bonita.
+
+{description}
+
+== Declare the interface in Bonita application descriptor
+
+The xref:applications:application-creation.adoc[Bonita application descriptor] is a file that let you describe all your applications.
+
+To declare a Bonita UI Designer Pro interface in it, edit it in Studio using the "source" tab.
+In this tab, you will be able to declare your application built and packaged with Bonita UI Designer Pro, using auto-completion with a "advancedApplication" XML tag.
```
Comment (csouchet) [inline modules/pages-and-forms/pages/pro-declare-interface-in-bonita.adoc:12]: **suggestion:** 
```suggestion
In this tab, you will be able to declare your application built and packaged with Bonita UI Designer Pro, using auto-completion with a `advancedApplication` XML tag.
```

File: modules/pages-and-forms/pages/pro-declare-interface-in-bonita.adoc [hunk 7]
```diff
@@ -0,0 +1,27 @@
+= Declare the interface in Bonita
+:page-aliases: ROOT:pro-declare-interface-in-bonita.adoc
+:description: This page defines the steps to declare your interface and display it in Bonita.
+
+{description}
+
+== Declare the interface in Bonita application descriptor
+
+The xref:applications:application-creation.adoc[Bonita application descriptor] is a file that let you describe all your applications.
```
Comment (csouchet) [inline modules/pages-and-forms/pages/pro-declare-interface-in-bonita.adoc:9]: **nitpick:** 
The link of `xref:applications:application-creation.adoc[Bonita application descriptor]` is not working.

File: modules/pages-and-forms/pages/pro-declare-interface-in-bonita.adoc [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+To declare a Bonita UI Designer Pro interface in it, edit it in Studio using the "source" tab.
```
Comment (csouchet) [inline modules/pages-and-forms/pages/pro-declare-interface-in-bonita.adoc:11]: **suggestion:** 
```suggestion
To declare a Bonita UI Designer Pro interface in it, edit it in Studio using the `source` tab.
```

File: modules/pages-and-forms/pages/pro-declare-interface-in-bonita.adoc [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+In this tab, you will be able to declare your application built and packaged with Bonita UI Designer Pro, using auto-completion with a "advancedApplication" XML tag.
+
+image::images/advanced-application-descriptor.png[Application Descriptor Editor]
+
+In this example, the "Advanced Overview" application is declared with the tag:
```
Comment (csouchet) [inline modules/pages-and-forms/pages/pro-declare-interface-in-bonita.adoc:16]: **suggestion:** 
```suggestion
In this example, the _Advanced Overview_ application is declared with the tag:
```

File: modules/pages-and-forms/pages/pro-declare-interface-in-bonita.adoc [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+
+[source,xml]
+----
+    <advancedApplication token="advanced" version="1.0" profile="User" state="ACTIVATED">
+        <displayName>Advanced Overview</displayName>
+    </advancedApplication>
+----
+
+There are fewer properties to set for an advanced application than for a legacy application, but the "displayName" is still mandatory.
```
Comment (csouchet) [inline modules/pages-and-forms/pages/pro-declare-interface-in-bonita.adoc:25]: **suggestion:** 
```suggestion
There are fewer properties to set for an advanced application than for a legacy application, but the `displayName` is still mandatory.
```

File: modules/pages-and-forms/pages/pro-declare-interface-in-bonita.adoc (hunk 9 above)
Comment (csouchet) [inline modules/pages-and-forms/pages/pro-declare-interface-in-bonita.adoc:16]: **nitpick:** 
For blind people, it's not clear whether to add this snippet code inside the `<applications></applications>` tag.

**Comment by jeromecambon** [comment]
This PR will be closed soon, it's replaced by https://github.com/bonitasoft/bonita-doc/pull/2714

**Pull request closed without merging**